mlrun/runtimes/function.py

```diff
@@ -363,14 +370,31 @@ def get_fullname(config, name, project, tag):
         return self.spec.command
 
     def deploy_step(
-        self, dashboard="", project="", models=None, env=None, tag=None, verbose=None
+        self,
+        dashboard="",
+        project="",
+        models=None,
+        env=None,
+        tag=None,
+        verbose=None,
+        use_db=True,
     ):
         models = {} if models is None else models
         name = "deploy_{}".format(self.metadata.name or "function")
         project = project or self.metadata.project
+        if models and isinstance(models, dict):
+            models = [{"name": k, "model_path": v} for k, v in models.items()]
+
+        if use_db:
+            hash_key = self.save(versioned=True, refresh=True)
+            url = "db://" + self._function_uri(hash_key=hash_key)
```

Comment:
@Hedingber this is not for end users, the right was is to use the DB, this is a backdoor for using the old way in case we run into issues w it, we have the same in jobs `as_step()` , once we see things are stable we should deprecate the `use_db` option.

Reply:
First of all `deploy_step` is fully user-facing, the fact you don't want users to use it, doesn't mean they wouldn't.
Second, also we, the developers need to be able to understand what is the meaning of an argument, if you can make it more clear by simple renaming, do it.
Third - if you won't leave some comment saying explictly that this should be deprecated when X happens, it would never be done cause simply no one except you know about this.
Bottomline:
I still suggest the rename, if you so insist leaving it as is, at least add a docstring (and actually anyway a docstring is needed here), if you plan to deprecate it, comment about it.

Reply:
done